File: Chapter 03/JS/section_3_3_2_ocp/after/passToStriker.js
/**
 * OCP SOLUTION: Open for Extension.
 * * ARCHITECTURE NOTE: This class proves that our system is "Open for extension." 
 * We added this brand new feature (a new play) simply by creating a new file. 
 * We didn't have to touch a single line of code in Midfielder.js to make 
 * this new behavior available!
 */
class PassToStriker {
    execute() {
        console.log("  [Action] Passing the ball to the striker!");
    }
}

module.exports = PassToStriker;

File: Chapter 03/Python/section_3_3_2_ocp/after/pass_to_striker.py
from .play import Play

"""
OCP SOLUTION: Open for Extension.
* ARCHITECTURE NOTE: This class is the proof that the system is "Open for 
extension." We added a brand new behavior (a new play) by simply creating 
a new file. We didn't have to change a single line of code in the 
Midfielder class to make this work!
"""
class PassToStriker(Play):
    def execute(self):
        print("  [Action] Passing the ball to the striker!")